sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -1067,90 +1083,115 @@ public Object call(SQLDatabase db) throws Exception{
+                                try {
+                                    PreparedAttachment pa = prepareAttachment(usa);
+                                    attachmentManager.addAttachment(db, pa, rev);
+                                } catch (Exception e) {
+                                    logger.log(Level.SEVERE, "There was a problem adding the " +
+                                                    "attachment "
+                                                    + usa + "to the datastore for document " + rev,
+                                            e);
+                                    throw e;
                                 }
                             }
-                        } else {
+                        }
+                    } else {
 
-                            try {
-                                if (preparedAttachments != null) {
-                                    for (String[] key : preparedAttachments.keySet()) {
-                                        String id = key[0];
-                                        String rev = key[1];
+                        try {
+                            if (preparedAttachments != null) {
+                                for (String[] key : preparedAttachments.keySet()) {
+                                    String id = key[0];
+                                    String rev = key[1];
+                                    try {
                                         BasicDocumentRevision doc = getDocumentInQueue(db, id, rev);
-                                        if(doc != null) {
-                                            for (PreparedAttachment att : preparedAttachments.get(key)) {
+                                        if (doc != null) {
+                                            for (PreparedAttachment att : preparedAttachments.get
+                                                    (key)) {
                                                 attachmentManager.addAttachment(db, att, doc);
                                             }
                                         }
+                                    } catch (DocumentNotFoundException e){
+                                        continue;
                                     }
                                 }
-                            } catch(Exception e){
-                                logger.log(Level.SEVERE, "There was a problem adding an " +
-                                         "attachment to " +
-                                        "the datastore, terminating replication", e);
-                                throw e;
                             }
-
+                        } catch (Exception e) {
+                            logger.log(Level.SEVERE, "There was a problem adding an " +
+                                    "attachment to " +
+                                    "the datastore, terminating replication", e);
+                            throw e;
```

Comment:
If I recall correctly, if you follow the stack up, the replication will get terminated when the exception reaches the `ReplicationStrategy` so both the transaction and replication should get aborted